src/cs-core/SolarSystem.cpp

```diff
@@ -29,10 +29,10 @@ namespace cs::core {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-SolarSystem::SolarSystem(std::shared_ptr<const Settings> const& settings,
-    std::shared_ptr<utils::FrameTimings> const&                 frameTimings,
-    std::shared_ptr<GraphicsEngine> const&                      graphicsEngine,
-    std::shared_ptr<TimeControl> const&                         timeControl)
+SolarSystem::SolarSystem(std::shared_ptr<Settings> const& settings,
+    std::shared_ptr<utils::FrameTimings> const&           frameTimings,
+    std::shared_ptr<GraphicsEngine> const&                graphicsEngine,
+    std::shared_ptr<TimeControl> const&                   timeControl)
```

Comment:
Should all be passed by value and moved.

Reply:
You're right but in order to avoid too many conflicts with the `feature/clang-tidy` I will do this after it got merged.